File: Modules/hospital/Patients/PatientsForm.cs
using Serene2.Modules.hospital.Patients.Enum;

namespace Serene2.hospital.Forms;

[FormScript("hospital.Patients")]
[BasedOnRow(typeof(PatientsRow), CheckNames = true)]
public class PatientsForm
{
    public string PatientName { get; set; }
    public int Age { get; set; }

    public DateTime DateOfBirth { get; set; }

    public GenderEnum Gender { get; set; }

    [Required]
    public int GovernateId { get; set; }
    public int CityId { get; set; }

    [TextAreaEditor(Rows = 5)]
    public string Address { get; set; }
    public int DoctorId { get; set; }

    public int Cost { get; set; }

    [DisplayName("Loyalty Years")]
    public int LoyaltyYears { get; set; }

    [DisplayName("Total Discount"),ReadOnly(true)]
    public int TotalDiscount { get; set; }
    
    [Required]
    
    public int TaskValue { get; set; }

}

File: Modules/ServerTypes/hospital/PatientsForm.ts
import { StringEditor, IntegerEditor, DateEditor, EnumEditor, LookupEditor, TextAreaEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { GenderEnum } from "../Modules/hospital.Patients.Enum.GenderEnum";

export interface PatientsForm {
    PatientName: StringEditor;
    Age: IntegerEditor;
    DateOfBirth: DateEditor;
    Gender: EnumEditor;
    GovernateId: LookupEditor;
    CityId: LookupEditor;
    Address: TextAreaEditor;
    DoctorId: LookupEditor;
    Cost: IntegerEditor;
    LoyaltyYears: IntegerEditor;
    TotalDiscount: IntegerEditor;
    TaskValue: IntegerEditor;
}

export class PatientsForm extends PrefixedContext {
    static readonly formKey = 'hospital.Patients';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PatientsForm.init)  {
            PatientsForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;
            var w2 = DateEditor;
            var w3 = EnumEditor;
            var w4 = LookupEditor;
            var w5 = TextAreaEditor;

            initFormType(PatientsForm, [
                'PatientName', w0,
                'Age', w1,
                'DateOfBirth', w2,
                'Gender', w3,
                'GovernateId', w4,
                'CityId', w4,
                'Address', w5,
                'DoctorId', w4,
                'Cost', w1,
                'LoyaltyYears', w1,
                'TotalDiscount', w1,
                'TaskValue', w1
            ]);
        }
    }
}

[GenderEnum]; // referenced types